### PR TITLE
Add devcontainer and initial CI/CD tools

### PR DESCRIPTION
**Pull request by iMicknl**

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	// "forwardPorts": [],
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	"postCreateCommand": "pip3 install --user -r requirements.txt && pre-commit install",
+	"customizations": {
+		"vscode": {
+			"extensions": [
+				"ms-vscode.cpptools-extension-pack",
+				"GitHub.copilot-chat"
```
Comment (jamesmyatt) [inline .devcontainer/devcontainer.json:20]: I also have these:

```
        "ms-python.python",
        "visualstudioexptteam.vscodeintellicode",
        "redhat.vscode-yaml",
        "editorconfig.editorconfig",
        "ESPHome.esphome-vscode",
  ```

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/python
+{
+	"name": "Python 3",
+	// Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+	"image": "mcr.microsoft.com/devcontainers/python:1-3.12-bullseye",
```
Comment (jamesmyatt) [inline .devcontainer/devcontainer.json:6]: Try `bookworm`? Also try `3.11`? I think that would match the esphome image

Comment (iMicknl) [reply .devcontainer/devcontainer.json:6]: I will give it a try! However on GitHub Actions (CI) it doesn't run the full Docker image, just a default runner with Python 3.12. I will see how it runs with 3.11.

Interesting that you don't face the compile issue, but I do face them on the latest ESPHome addon in HaOS and on my dev environments..

**Comment by iMicknl** [comment]
Replaced by #90

**Pull request closed without merging**